### PR TITLE
Implement probabilistic damage rounding option and provide Russian translation

### LINKED ISSUE #187: Enable tools.damageMultiplicand property to be float/double.
### Describe the feature

Having a penalty for using whole tree breaking at once is a nice bonus, but only allowing integers feels too much for me: I want tools to damage only a bit faster, not break twice or thrice as quickly. Since damage inflicted upon the tool is required to be an integer, you can ```round```, ```floor```, or ```ceil``` the amount of damage to get an integer, and expose rounding option to the end user.

```tools.damageRounding: (round|floor|ceil|random)``` or something would be a nice addition. I'd love to set it to 1.2 or 1.4, ceil.

```random``` can be the same thing as with XP orbs: if the number is not an integer, you give the integer to the user and the remainder is used as a percentage of chance that they will receive another orb. Or, in this case, a damage point.

For example, 7.35 damage would translate to 65% of getting 7 damage and 35% of getting 8 damage.

### PR DESCRIPTION
**Pull request by MobButcher**
As discussed in #187, this PR would add a probabilistic damage rounding option that works like XP rounding in furnaces.

Also this PR includes translations to Russian, with separate commits for the rest of the mod and probabilistic damage rounding option translations. Hope you'll accept this pull request and that it would be helpful for you. ^^

P.S.: I've noticed your signing key is outdated. Consider generating a new one.

### CODE REVIEW

**Comment by MobButcher** [comment]
Sorry for force-pushes, there were some issues I didn't want to hang around in follow-up commits, especially before you merged this branch. I've double-checked that everything works on Forge too, which I didn't do before creating a pull request. My bad. But now it's better, I suppose.

File: fabric/src/main/java/fr/raksrinana/fallingtree/fabric/tree/breaking/ToolDamageHandler.java
Comment (Rakambda) [inline fabric/src/main/java/fr/raksrinana/fallingtree/fabric/tree/breaking/ToolDamageHandler.java]: Probably can use `damage` instead of calculating the floor a second time.
```suggestion
		double probability = rawDamage - damage;
```

File: forge/src/main/java/fr/raksrinana/fallingtree/forge/tree/breaking/ToolDamageHandler.java
Comment (Rakambda) [inline forge/src/main/java/fr/raksrinana/fallingtree/forge/tree/breaking/ToolDamageHandler.java]:
```suggestion
		double probability = rawDamage - damage;
```

**Review by Rakambda — CHANGES_REQUESTED** [review]

**Comment by Rakambda** [comment]
The force pushes are fine. The thing is that I rebased the 187 branch and force pushed it. So You're kinda detached from it now I guess (hence the merge conflict).
Will have to find a way to fix it. Maybe the easiest is to do a temp branch locally starting from the new head of 187, and cherry pick the commits you did. Then push that new temp branch to your `187-probabilistic-damage`.

File: fabric/src/main/java/fr/raksrinana/fallingtree/fabric/tree/breaking/ToolDamageHandler.java
Comment (MobButcher) [reply fabric/src/main/java/fr/raksrinana/fallingtree/fabric/tree/breaking/ToolDamageHandler.java]: I decided to go against it because of multiple type recastings. It might introduce discrepancies and errors on conversion. Although I don't know Java well enough to know whether it's actually the case, so if you believe there can't be any errors, second floor calculation is unnecessary indeed.

**Comment by MobButcher** [comment]
> The force pushes are fine. The thing is that I rebased the 187 branch and force pushed it. So You're kinda detached from it now I guess (hence the merge conflict).
> Will have to find a way to fix it. Maybe the easiest is to do a temp branch locally starting from the new head of 187, and cherry pick the commits you did. Then push that new temp branch to your `187-probabilistic-damage`.

GitHub says this branch has no conflicts with the base branch. Although maybe those commits would mess up when merging with `1.17.1` branch? Is there anything I could do with my branch? Maybe rebase it onto the same point you have?

File: fabric/src/main/java/fr/raksrinana/fallingtree/fabric/tree/breaking/ToolDamageHandler.java
Comment (Rakambda) [reply fabric/src/main/java/fr/raksrinana/fallingtree/fabric/tree/breaking/ToolDamageHandler.java]: There can be some errors when doing this, but it's tiny errors that don't really have much impact for us. Like if there's a 0.001 error in the conversion that isn't too much of a problem, we're not dealing with money or sensitive data.

Also floor performs quite a lot under the hood I think it's less costly to have it convert `int -> double` again rather than redoing the floor.

As we know we're dealing with positive values we could also avoid the floor altogether and do more "direct" operations:
```java
int damage = (int) rawDamage; //Casting to int rounds towards zero, while floor rounds towards negative infinity. As we have positive values, casting does the same
double probability = rawDamage % 1; // Will have to check this though
```

I let it up to you if wanna do the change or not. The performance doesn't really matter actually as this method isn't called often. Just if you wanna keep the double, put it at least in a variable to avoid performing all the floor calculations twice.

**Comment by Rakambda** [comment]
image

The part in read is actually the same. Would need to not depend on old commit (before i rebased), and have instead the orange path.
If you're not too much at ease with git, you can allow the edit by maintainers on the PR and i'll see if I can fix it.

What I explained above would be to basically put yourself back at the new HEAD of my 187 (last blue dot), create a new local branch (temporary), and then cherry pick your commits (all the green dots). When it's done you should have a straight line of green dots after my blue ones. Then the new temporary branch can be force pushed as your 187-proba branch.

**Comment by MobButcher** [comment]
Edits by maintainers are allowed. What's the tool you're using to visualize this? Because for some reason I can't see anything you're doing there. Unless your rebase is on local and isn't on GitHub yet, because I'm updating my fork upstream. ...unless I shouldn't do that.

You know what? I'll just leave `187-probabilistic-damage` alone and try to work on something else. Sorry for making this a mess.

**Comment by Rakambda** [comment]
It's ok don't worry. I'll try to fix that up later.

And the graph is seen here: https://github.com/RakSrinaNa/FallingTree/network

**Review by Rakambda — APPROVED** [review]

**Comment by Rakambda** [comment]
Grr, as I cherrypicked the commits I get it lost the fact you did the commits. If you wanna appear in the contributors list you can make a change in the changelog on the 1.17.1 branch.

Thanks for this PR tho 😉

**Pull request merged**
